src/HYPRE.jl

```diff
@@ -512,42 +540,49 @@ function Internals.copy_check(dst::HYPREVector, src::PVector)
     if il_dst != il_src && iu_dst != iu_src
         # TODO: Why require this?
         throw(ArgumentError(
-            "row owner mismatch between dst ($(il_dst:iu_dst)) and src ($(il_dst:iu_dst))"
+            "row owner mismatch between dst ($(il_dst:iu_dst)) and src ($(il_src:iu_src))"
         ))
     end
 end
 
 # TODO: Other eltypes could be support by using a intermediate buffer
-function Base.copy!(dst::PVector{HYPRE_Complex}, src::HYPREVector)
```

Comment:
Is this valid? Note that ccall doesnt convert IIRC.

Reply:
I think you're right that the type definition is now too wide. I removed the parametric dispatch since PArray now has a much more complicated type signature. I will see if I can't find a definition that works for the backends but still limits to HYPRE_Complex eltype.